airflow/www/api/experimental/endpoints.py

```diff
@@ -74,8 +75,12 @@ def trigger_dag(dag_id):
 
             return response
 
+    replace_microseconds = True
+    if 'replace_microseconds' in data:
+        replace_microseconds = to_boolean(data['replace_microseconds'])
+
     try:
-        dr = trigger.trigger_dag(dag_id, run_id, conf, execution_date)
+        dr = trigger.trigger_dag(dag_id, run_id, conf, execution_date, replace_microseconds)
```

Comment:
This would fit the goal, but I feel there is a better way of expressing this.

If you follow all the code through I think what we should do is: if an explicit timestamp is passed in, we should use that exactly. If one is not passed then the current default of "generate the execution date from now(), but truncate it to nearest whole seconds" come in to play.

I think that would lead to a clearer to understand API.

WDYT?

Reply:
yeah that seems much nicer.  I felt like passing `'replace_microseconds': 'false'` in the request body felt a bit odd, but didn't want to make _too_ big of a change to start with.  I'll make the changes in the morning

Reply:
upon second thought... is there a particular reason that the time _needs_ to be truncated?  having a default behaviour so limiting seems like a bug

Reply:
I found [this commit](https://github.com/apache/airflow/commit/b7c828bf094d3aa1eae310979a82addf7e423bb0) from Feb 2017 that mentions why microseconds were replaced at that time.  I would like to think that isn't still relevant, but I'm not sure to be honest.

Reply:
It doesn't seen needed, other than it's sometimes "nice" to see whole seconds. For the API it is probably more sensible to default to un-trucated time, but this would be potentially "breaking" change, so we need to think about how we upgrade existing users.

For the 1.10 release series I would probably suggest that the default shouldn't change, just in case someone is relying on it.

If you could do this as two PRs (both against master) the first adding the behavour we talked about, but not changing the deafult behaviour (which we can get in to the next 1.10.x release, I will cherry-pick the merged PR back to the relase branch) and then once that is merged a second one that changes the default to be as you describe.

How does that sound?

Reply:
sounds great, will do!